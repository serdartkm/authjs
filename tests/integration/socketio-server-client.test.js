const path = require('path')
require('dotenv').config({ path: path.join(__dirname, `./.env`) })//
const jwt = require('jsonwebtoken')

//client
import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'  //client
import {mount,configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ReactTestRenderer from 'react-test-renderer'

import io from "socket.io-client"; //client
import MessagingModuleSocket from '../../rtcjs/messaging-module-socket' //client
const serverURL = "http://localhost:3000/" //client

configure({adapter:new Adapter()})

describe("INTEGRATION TESTING ", () => {

  describe("socketio server and client messaging", () => {//

    it("client sever connection established with required params", async (done) => {
      const tokenMario = await jwt.sign({ data: "mario" }, "secret", { expiresIn: '1h' })
      const tokenDragos = await jwt.sign({ data: "dragos" }, "secret", { expiresIn: '1h' })
      const socketServer = require('../../rtcjs/socketio-server')({})
      const marioClient = io(tokenMario, "one");
      const dragosClient = io(tokenDragos, "two");

      marioClient.onconnection((client) => {
        debugger
        try {

          (()=>{

            debugger
          //  const { getByText, getByPlaceholderText } = render(<MessagingModuleSocket id={1} name="mario" targetName="dragos" socket={client} />)
             const testRenderer =ReactTestRenderer.create(<MessagingModuleSocket id={1} name="mario" targetName="dragos" socket={client} />)
            //fireEvent.change(getByPlaceholderText("Enter message text"), { target: { value: "Hello My Dear" } })
            client.on("text_message",(data)=>{
              testRenderer.update()
              debugger

               const jsonformat=     testRenderer.toJSON()
             // expect(getByText("Hello My Dear")).toBeVisible()
  
              done()
            })
          })()
      
        } catch (error) {
          debugger
        }
      })
      dragosClient.onconnection((client) => {
        try {

          (()=>{
      
            debugger
            const { getByTestId  } = render(<MessagingModuleSocket id={2} name="dragos" targetName="mario" socket={client} />)
            fireEvent.change(getByTestId(`message${2}`), { target: { value: "Hello My Dear" } })
            fireEvent.click(getByTestId(`sendMessage${2}`))
          })()
        
        } catch (error) {
          debugger
        }
        // expect(getByText("Hello My Dear")).toBeVisible()
      })

      socketServer.onconnection((socket) => {
        socket.join("mario")
        debugger


      })

 
    })


  })
})

