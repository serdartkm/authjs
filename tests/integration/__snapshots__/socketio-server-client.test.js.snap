// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messaging-module-socket and nodejs-socketio-text-chat positive testing client sever connection established with required params 1`] = `
<div>
  <div
    style="height: 96vh; width: 100%;"
  >
    <div
      name="msgViewScroller"
      style="background-color: rgb(237, 239, 242); overflow: auto; width: 100%; height: 100%;"
    >
      <div>
        <div
          style="display: flex;"
        >
          <div
            style="flex: 1;"
          >
            <hr />
          </div>
          <div>
            6/15/2015
          </div>
          <div
            style="flex: 1;"
          >
            <hr />
          </div>
        </div>
        <div
          style="display: flex; justify-content: flex-start;"
        >
          <div
            style="display: flex; align-items: center;"
          >
            <div
              style="height: 30px; width: 40px; padding: 3px; border-radius: 30px; border-style: solid; border-width: 2px; display: flex; justify-content: center; align-items: center; color: rgb(0, 150, 136); border-color: #80cbc4;"
            >
              <div>
                D
              </div>
            </div>
            <div
              style="background-color: rgb(255, 236, 179); padding: 5px; margin: 2px; border-radius: 15px; border-color: #9e9e9e; border-style: solid; border-width: 2px; max-width: 100%; word-wrap: break-word; word-break: break-all; min-width: 30%;"
            >
              <div>
                Hello My Dear
              </div>
              <div
                style="font-size: 10px; padding-top: 2px; text-align: end;"
              >
                <i
                  style="background-color: rgb(239, 235, 233);"
                >
                  3:12:05 AM
                </i>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="display: flex;"
    >
      <input
        data-testid="message1"
        name="message"
        placeholder="Enter message text"
        style="flex: 1;"
        type="text"
      />
      <div
        style="display: flex;"
      >
        <button
          data-testid="sendMessage1"
          disabled=""
          style="margin-left: 2px; width: 100%;"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
